File: packages/kotlin/src/android/src/main/java/alphaTab/core/Console.kt
package alphaTab.core

internal open class Console {
    public open fun debug(format: String, vararg details: Any?) {
        val message = if (details.isNotEmpty()) "$format,${details.joinToString(",")}" else format
        println("[AlphaTab Debug] $message")
    }

    public open fun warn(format: String, vararg details: Any?) {
        val message = if (details.isNotEmpty()) "$format,${details.joinToString(",")}" else format
        println("[AlphaTab Warn] $message")
    }

    public open fun info(format: String, vararg details: Any?) {
        val message = if (details.isNotEmpty()) "$format,${details.joinToString(",")}" else format
        println("[AlphaTab Info] $message")
    }

    public open fun error(format: String, vararg details: Any?) {
        val message = if (details.isNotEmpty()) "$format,${details.joinToString(",")}" else format
        println("[AlphaTab Error] $message")
    }
}

File: packages/csharp/src/AlphaTab/Core/Console.cs
using System.Diagnostics;

namespace AlphaTab.Core;

internal class Console
{
    public virtual void Debug(string format, object?[]? details)
    {
        var message = details != null ? string.Format(format, details) : format;
        Trace.WriteLine(message, "AlphaTab Debug");
    }

    public virtual void Warn(string format, object?[]? details)
    {
        var message = details != null ? string.Format(format, details) : format;
        Trace.WriteLine(message, "AlphaTab Warn");
    }

    public virtual void Info(string format, object?[]? details)
    {
        var message = details != null ? string.Format(format, details) : format;
        Trace.WriteLine(message, "AlphaTab Info");
    }

    public virtual void Error(string format, object?[]? details)
    {
        var message = details != null ? string.Format(format, details) : format;
        Trace.WriteLine(message, "AlphaTab Error");
    }
}
